### PR TITLE
Cleanup feature model and all related to feature enabled implementation

### PR DESCRIPTION
**Pull request by ferryphang**
This PR resolves the issue described in #657 

- I delete the method and logic inside the Feature.rb
- Remove all if else statement that use the Feature.enabled?

## Pull request checklist

- [ ] My code contains tests covering the code I modified
- [x] I linted and tested the project with `bin/check`
- [ ] I added significant changes and product updates to the [changelog](CHANGELOG.md)

### CODE REVIEW

File: app/models/feature.rb
Comment (joemasilotti) [inline app/models/feature.rb:15]: Sorry, I wrote this issue before this feature was added. Can you please keep this case and the code around it?

File: app/views/shared/_main.html.erb
Comment (joemasilotti) [inline app/views/shared/_main.html.erb:4]: IIRC, we can now remove the parameter and code in `Developers::NewFieldsComponent` to check for "enabled" now, too.

File: app/controllers/pricing_controller.rb
Comment (joemasilotti) [inline app/controllers/pricing_controller.rb:11]: We can now remove this entire method and let the `#show` method render `app/views/pricing/show.html.erb`. This means removing all of the v1 and v2 templates/partials and moving the v3 stuff to the "root" level of that directory.

**Review by joemasilotti — CHANGES_REQUESTED** [review]
Thanks for taking a pass! I left a few comments and requests – let me know if you have any questions.

File: app/controllers/pricing_controller.rb
Comment (ferryphang) [reply app/controllers/pricing_controller.rb]: I have move the v3 view to root and improve the controller

File: app/views/shared/_main.html.erb
Comment (ferryphang) [reply app/views/shared/_main.html.erb:4]: Okay i have remove the params for "enabled" and the test for that

File: app/models/feature.rb
Comment (ferryphang) [reply app/models/feature.rb]: Yup please check if its alright

**Review by joemasilotti — CHANGES_REQUESTED** [review]
Looks great! Thanks for taking the time to go through all the pricing files.

I made one request that can be applied to a few other partials, too. Once that's done this should be good to merge.

File: app/views/pricing/show.html.erb
Comment (ferryphang) [reply app/views/pricing/show.html.erb:12]: Opps forgot about that

**Comment by joemasilotti** [comment]
Looks great! I pushed a merge commit and will merge when CI passes.

**Pull request merged**
